plugins/welcome-plugin/src/welcome-plugin.ts

```diff
@@ -0,0 +1,142 @@
+
+import * as theia from '@theia/plugin';
+import * as path from 'path';
+import { WelcomePage } from './welcome-page';
+
+
+async function getHtmlForWebview(context: theia.PluginContext): Promise<string> {
+
+    // Local path to main script run in the webview
+    const scriptPathOnDisk = theia.Uri.file(path.join(context.extensionPath, 'resources', 'welcome-page.js'));
+    // And the uri we use to load this script in the webview
+    const scriptUri = scriptPathOnDisk.with({ scheme: 'theia-resource' });
+
+
+    // Local path to main script run in the webview
+    const cssPathOnDisk = theia.Uri.file(path.join(context.extensionPath, 'resources', 'welcome-page.css'));
+    // And the uri we use to load this script in the webview
+    const cssUri = cssPathOnDisk.with({ scheme: 'theia-resource' });
+
+
+    const welcomePage = new WelcomePage(context);
+    const rendering = await welcomePage.render(context);
+    return `<!DOCTYPE html>
+            <html lang="en">
+            <head>
+                <meta charset="UTF-8">
+                <meta http-equiv="Content-Security-Policy" content="default-src 'none'; font-src 'self'; style-src 'self' 'unsafe-inline'; img-src 'self'; script-src 'unsafe-inline' 'self' ;">
+                <meta name="viewport" content="width=device-width, initial-scale=1.0">
+                <link rel="stylesheet" type="text/css" href="${cssUri}">
+                </style>
+                <title>Welcome Page</title>
+            </head>
+
+            <body>
+              ${rendering}
+              <script src="${scriptUri}"></script>
+            </body>
+            </html>`;
+
+}
+
+export async function doStart(context: theia.PluginContext): Promise<void> {
+    addPanel(context);
+    handleReadmeFiles(context);
+}
+
+
+// Open Readme file is there is one
+export async function handleReadmeFiles(context: theia.PluginContext): Promise<void> {
+
+    const roots: theia.WorkspaceFolder[] | undefined = await theia.workspace.workspaceFolders;
+    // In case of only one workspace
+    if (roots && roots.length === 1) {
+        const children = await theia.workspace.findFiles('README.md', 'node_modules/**', 1);
+        const updatedChildren = children.filter((child: theia.Uri) => {
+            if (child.fsPath.indexOf('node_modules') === -1) {
+                return child;
+            }
+
+        })
+
+        if (updatedChildren.length >= 1) {
+            var openPath = theia.Uri.parse(updatedChildren[0] + '?open-handler=code-editor-preview');
+            const doc: theia.TextDocument | undefined = await theia.workspace.openTextDocument(openPath);
+            if (doc) {
+                theia.window.showTextDocument(doc);
+            }
+        }
+    }
+}
+
+export async function addPanel(context: theia.PluginContext): Promise<void> {
+    // Track currently webview panel
+    let currentPanel: theia.WebviewPanel | undefined = undefined;
+
+    if (currentPanel) {
```

Comment:
`currentPanel` is always `undefined` here, isn't it?

Reply:
@benoitf I didn't mean here to remove declaration for `currentPanel`. You can remove `if` statement because `currentPanel` is initialized as `undefined`, and `return` will never happen there.

Reply:
I hope I fixed it now :-)